crates/pixi_manifest/src/s3.rs

```diff
@@ -0,0 +1,14 @@
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+/// Custom S3 configuration
+#[derive(Debug, Clone, PartialEq, Serialize, Eq, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct S3Options {
```

Comment:
currently we require to set all three values at once. At some point we might want to investigate defaults in the protocol but since it would be backwards compatible, i think this is fine for now